### PR TITLE
Better error reporting for unsupported case like max(x)-min(y)

### PR DESCRIPTION
**Pull request by galkk**
*Issue #, if available:* #194, #234 

*Description of changes:*
Better error reporting for such queries

```
POST /_opendistro/_sql/_explain
{
  "query":"""SELECT DestCountry, dayOfWeek, max(FlightDelayMin) - min(FlightDelayMin) + avg(FlightDelayMin)
    FROM kibana_sample_data_flights
    GROUP BY DestCountry, dayOfWeek
  """
```

returns now 

```
{
  "error": {
    "reason": "There was internal problem at backend",
    "details": "The complex aggregate expressions are not implemented yet: MAX(FlightDelayMin) - MIN(FlightDelayMin)",
    "type": "SqlFeatureNotImplementedException"
  },
  "status": 503
}
```

instead of non-sensical:

```
{
  "error": {
    "reason": "Invalid SQL query",
    "details": "",
    "type": "NullPointerException"
  },
  "status": 400
}
```

and another example:
````
                        "SELECT DestCountry, dayOfWeek, log(max(FlightDelayMin))" +
                        "   FROM kibana_sample_data_flights\n" +
                        "   GROUP BY DestCountry, dayOfWeek\n"));
```

witll throw

                  "Function calls of form 'log(MAX(...))' are not implemented yet");


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/SqlParserTest.java [hunk 1]
```diff
@@ -30,18 +30,20 @@
...
 import com.amazon.opendistroforelasticsearch.sql.query.maker.QueryMaker;
 import com.amazon.opendistroforelasticsearch.sql.query.multi.MultiQuerySelect;
 import com.amazon.opendistroforelasticsearch.sql.util.CheckScriptContents;
+import org.elasticsearch.client.Client;
 import org.elasticsearch.index.query.BoolQueryBuilder;
 import org.elasticsearch.search.builder.SearchSourceBuilder.ScriptField;
-import org.junit.Assert;
-import org.junit.BeforeClass;
-import org.junit.Test;
+import org.junit.*;
```
Comment (dai-chen) [inline src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/SqlParserTest.java]: I assume this violates our Checkstyle and should fail the GitHub Action check :)

Comment (dai-chen) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/SqlParserTest.java]: It seems passed though?

Comment (galkk) [reply src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/parser/SqlParserTest.java]: yeah, seems like it is warning :)

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java [hunk 2]
```diff
@@ -317,6 +319,10 @@ public MethodField makeMethodField(String name, List<SQLExpr> arguments, SQLAggr
             } else if (object instanceof SQLCastExpr) {
                 String scriptCode = new CastParser((SQLCastExpr) object, alias, tableAlias).parse(false);
                 paramers.add(new KVValue("script", new SQLCharExpr(scriptCode)));
+            } else if (object instanceof SQLAggregateExpr) {
```
Comment (dai-chen) [inline src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java]: Does this also cover other cases such as `MAX(age + 1)`, `MAX(LOG(age))`? If not, any chance to take care of them too?

**Comment by chloe-zh** [comment]
I think the code changes for now covers more than cases like `max(x) - min(y)`. It seems the `object` is always the left one of a binary operation, so it can cover the cases as long as it looks like `max(x) + y`. Could you implement the cases of `x + min(y)`-like in this PR as well?

**Comment by galkk** [comment]
@chloe-zh 

The case that you've mentioned already works

```
POST /_opendistro/_sql/_explain
{
  "query":"""SELECT 1+max(FlightDelayMin)
    FROM kibana_sample_data_flights
    GROUP BY DestCountry, dayOfWeek
  """
}


{
  "error": {
    "reason": "There was internal problem at backend",
    "details": "The complex aggregate expressions are not implemented yet: 1 + MAX(FlightDelayMin)",
    "type": "SqlFeatureNotImplementedException"
  },
  "status": 503
}
```

File: src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java (hunk 2 above)
Comment (galkk) [reply src/main/java/com/amazon/opendistroforelasticsearch/sql/parser/FieldMaker.java]: > MAX(age + 1),

This is quite different case, out of scope of this PR

> MAX(LOG(age))
surprisingly, this is **almost** working, I think it should be just fixed

**Pull request merged**